conanfile.py

```diff
@@ -0,0 +1,115 @@
+    generators = "CMakeDeps"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+
+    default_options = {"shared": False, "fPIC": True,}
+
+    exports_sources = shared_sources
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["MINIFI_BUILD_CONAN_PACKAGE"] = "ON"
+        tc.variables["MINIFI_LIBCURL_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_OPENSSL_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_ZLIB_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_CIVETWEB_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_LIBXML2_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_CATCH2_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_FMT_SOURCE"] = "CONAN"
+        tc.variables["MINIFI_SPDLOG_SOURCE"] = "CONAN"
+
+        tc.variables["SKIP_TESTS"] = "OFF"
+        tc.variables["ENABLE_OPENWSMAN"] = "ON"
+        tc.variables["ENABLE_CIVET"] = "ON"
+        tc.variables["ENABLE_EXPRESSION_LANGUAGE"] = "ON"
+        tc.variables["ENABLE_BZIP2"] = "ON"
+        tc.variables["ENABLE_ROCKSDB"] = "ON"
+        tc.variables["BUILD_ROCKSDB"] = "ON"
+
+        tc.variables["ENABLE_LIBARCHIVE"] = "OFF"
+        tc.variables["ENABLE_AWS"] = "OFF"
+
+        tc.variables["ENABLE_OPC"] = "OFF"
+
+        if self.settings.os == "Windows":
+            tc.variables["ENABLE_WEL"] = "OFF"
+            tc.variables["ENABLE_PDH"] = "OFF"
+            tc.variables["ENABLE_SMB"] = "OFF"
+        elif self.settings.os == "Linux":
+            tc.variables["ENABLE_SYSTEMD"] = "OFF"
+            tc.variables["ENABLE_PROCFS"] = "OFF"
+
+        tc.variables["ENABLE_LZMA"] = "OFF"
+        tc.variables["ENABLE_GPS"] = "OFF"
+        tc.variables["ENABLE_COAP"] = "OFF"
+        tc.variables["ENABLE_SQL"] = "OFF"
+        tc.variables["ENABLE_MQTT"] = "OFF"
+        tc.variables["ENABLE_PCAP"] = "OFF"
+        tc.variables["ENABLE_LIBRDKAFKA"] = "OFF"
+        tc.variables["ENABLE_LUA_SCRIPTING"] = "OFF"
+        tc.variables["ENABLE_PYTHON_SCRIPTING"] = "OFF"
+        tc.variables["ENABLE_SENSORS"] = "OFF"
+        tc.variables["ENABLE_USB_CAMERA"] = "OFF"
+        tc.variables["ENABLE_OPENCV"] = "OFF"
+        tc.variables["ENABLE_BUSTACHE"] = "OFF"
+        tc.variables["ENABLE_SFTP"] = "OFF"
+        tc.variables["ENABLE_AZURE"] = "OFF"
+        tc.variables["ENABLE_ENCRYPT_CONFIG"] = "OFF"
+        tc.variables["ENABLE_SPLUNK"] = "OFF"
+        tc.variables["ENABLE_ELASTICSEARCH"] = "OFF"
+        tc.variables["ENABLE_GCP"] = "OFF"
+        tc.variables["ENABLE_KUBERNETES"] = "OFF"
+        tc.variables["ENABLE_TEST_PROCESSORS"] = "OFF"
+        tc.variables["ENABLE_PROMETHEUS"] = "OFF"
+        tc.variables["ENABLE_GRAFANA_LOKI"] = "OFF"
+        tc.variables["ENABLE_GRPC_FOR_LOKI"] = "OFF"
+        tc.variables["ENABLE_CONTROLLER"] = "OFF"
+
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
```

Comment:
Here is conan's way of packaging assets after building MiNiFi C++ where we can tell conan to collect the headers, library (static, dynamic, etc), app binaries, etc @szaszm @lordgamez 

In CMake standalone, we would build MiNiFi C++ and then run `make package` to create MiNiFi binary tarball.

However, by adding this ConanFile package method, we are able to run `conan create minifi-cpp ...`, which can then be uploaded to a conan repository like conancenter, jfrog, github, etc. Whether we now use the local minifi-cpp conan package or remote one, now we can easily integrate minifi-cpp into other C++ software infrastructures through their conanfile.py.